### DISCUSSION TITLE
Unable to get PostData from IRequest

### DISCUSSION THREAD

**Discussion by aiqinxuancai** [cefsharp/CefSharp#4784] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

123.0.60

### Operating System

Windows 10

### Architecture

x64

### .Net Version

.Net8.0

### Implementation

WPF

### Reproduction Steps

I have a program embedded with CefSharp, which I use to intercept some POST requests and process them. It has been working well for a long time, but recently it seems that the website has been updated. Without any updates or changes to my program, I am unable to retrieve the data from IRequest's PostData.
This is somewhat strange, as not all PostData cannot be retrieved, only a specific frame request on this website. In the debugging interface, I can see the FormData submitted by the request, but in the code, I cannot retrieve the PostData.

I tried testing with the simplest code possible, and the result was the same.

```csharp
public class CustomRequestHandler : RequestHandler
{
    protected override IResourceRequestHandler GetResourceRequestHandler(IWebBrowser chromiumWebBrowser, IBrowser browser, IFrame frame, IRequest request, bool isNavigation, bool isDownload, string requestInitiator, ref bool disableDefaultHandling)
    {
        if (request.Method == "POST")
        {
            var postData = request.PostData;
            if (postData != null)
            {
                var elements = postData.Elements;

                foreach (var element in elements)
                {
                    if (element.Type == PostDataElementType.Bytes)
                    {
                        var body = Encoding.UTF8.GetString(element.Bytes);
                        Debug.WriteLine("PostData#2：" + body);
                    }
                }
            }
        }

        return base.GetResourceRequestHandler(chromiumWebBrowser, browser, frame, request, isNavigation, isDownload, requestInitiator, ref disableDefaultHandling);
    }
}
```
In Debuging.
![QQ截图20240418141536](https://github.com/cefsharp/CefSharp/assets/4475018/b82f2d8f-62f3-4dcb-ae62-78c8a5860767)

In Code.
![QQ截图20240418142943](https://github.com/cefsharp/CefSharp/assets/4475018/1b67a7a6-7afe-484d-be9c-9860cbcd5e9f)

### Expected behavior

Correctly obtaining PostData.

### Actual behavior

Unable to get PostData

### Regression?

Yes, I encountered this issue in version 95 of CefSharp, and I can still reproduce it after upgrading to version 123.

### Known Workarounds

No

### Does this problem also occur in the CEF Sample Application

Yes using WPF/OffScreen command line args

**Comment by amaitland** [comment]
What is the value of https://cefsharp.github.io/api/118.6.x/html/P_CefSharp_IPostData_HasExcludedElements.htm

If the value is true then it's likely what you are seeing is expected. If it's false then you'll need to raise an issues at https://github.com/chromiumembedded/cef/issues

`CefSharp` is just a wrapper around the `Chromium Embedded Framework`.

**Reply by aiqinxuancai** [reply]
> What is the value of https://cefsharp.github.io/api/118.6.x/html/P_CefSharp_IPostData_HasExcludedElements.htm
> 
> If the value is true then it's likely what you are seeing is expected. If it's false then you'll need to raise an issues at https://github.com/chromiumembedded/cef/issues
> 
> `CefSharp` is just a wrapper around the `Chromium Embedded Framework`.

I have found the reason. By setting a breakpoint in Chrome's debugging tool and looking at the parameters of `window.fetch`, I noticed that the data type of `PostData` is Blob, with an empty type, but it does have data in `size`. By using the code `const text = await body.text();` in JavaScript, you can get the parameters.
It seems that in CEF or CefSharp, it is can't to support this type of data. Currently, I am using injected JavaScript code to get this `PostData`, and it works well.
![image](https://github.com/cefsharp/CefSharp/assets/4475018/e05c2f53-dd0f-46a3-b876-59a33a739507)

**Reply by amaitland** [reply]
I suspect this is a CEF limitation, you can ask at https://www.magpcss.org/ceforum/index.php for clarification. See if the CEF maintainer thinks this is fixable.
